### DISCUSSION TITLE
Make only certain popup menus (GTK3/4) have rounded corners

### DISCUSSION THREAD

**Discussion by dglava** [yshui/picom#1304] [Q&A]
I wanted to add rounded corners, mainly as a fix for certain corners having black parts (for example in Nemo). It looks fine and works when I add them. Now I wanted to only apply them to popup menus because that's where they appear.

I added them like this:
```
corner-radius = 10

# Exclude everything but the popup menus
rounded-corners-exclude = [
  "!window_type = 'popup_menu'",
];
```
It works, but as a side-effect it applied it to some items I didn't know were popup menus. Most prominently the Openbox root menu. I tried to add Openbox windows to the exclusion list, but that somehow breaks the previous rule. Using the config below doesn't give me any rounded corners at all.

```
rounded-corners-exclude = [
  "!window_type = 'popup_menu'",
  "'class_g = 'Openbox'"
];
```

Does anyone have an idea how to approach this?

**Comment by absolutelynothelix** [comment]
you have a syntax error in the second rule, there's an unmatched single quote.

**Reply by dglava** [reply]
Thanks for pointing it out. I have corrected it to:
```
corner-radius = 8

rounded-corners-exclude = [
  "!window_type = 'popup_menu'",
  "class_g = 'Openbox'"
];
```
The Openbox root menu still has rounded corners.
